starfish/core/pipeline/algorithmbase.py

```diff
@@ -21,29 +18,33 @@ def run_with_logging(func):
         This method extends each pipeline component.run() method to also log itself and
         runtime parameters to the IntensityTable and ImageStack objects. There are two
         scenarios for this method:
-            1.) Filtering/ApplyTransform:
-                    Imagestack -> Imagestack
-            2.) Spot Detection:
-                    ImageStack -> IntensityTable
-                    ImageStack -> [IntensityTable, ConnectedComponentDecodingResult]
-            TODO segmentation and decoding
         """
         @functools.wraps(func)
         def helper(*args, **kwargs):
             result = func(*args, **kwargs)
-            # Scenario 1, Filtering, ApplyTransform
-            if isinstance(result, ImageStack):
-                result.update_log(args[0])
-            # Scenario 2, Spot detection
-            elif isinstance(result, tuple) or isinstance(result, IntensityTable):
-                if isinstance(args[1], ImageStack):
-                    stack = args[1]
-                    # update log with spot detection instance args[0]
-                    stack.update_log(args[0])
-                    # get resulting intensity table and set log
-                    it = result
-                    if isinstance(result, tuple):
-                        it = result[0]
-                    it.attrs[STARFISH_EXTRAS_KEY] = LogEncoder().encode({LOG: stack.log})
+            if result is not None:
+                method_class_str = str(args[0].__class__)
+                if 'ApplyTransform' in method_class_str or 'Filter' in method_class_str:
```

Comment:
It's plausible that we have mechanisms that take multiple loggable objects.

It might be a good idea to have a base `ProvenanceTracked` class that all objects (`ImageStack`, `IntensityTable`, etc.) inherit, and then to walk through all the arguments to find objects that extend `ProvenanceTracked`.  Then scrape the logs from those objects and build up a provenance graph.

Reply:
this seems like a larger refactor, like maybe something post spot finding stuff

Reply:
Sure, can you tack that onto the logging epic?

There are a lot of gaps in the logging -- for instance, ApplyTransform plausibly should log the origin of the warping data.  Also can be covered afterwards.